### PR TITLE
ENH: Added __repr__ for Figure

### PR DESCRIPTION
**Pull request by richardjgowers**
## PR Summary

Adds a better repr to Figure
Eg:
```
<Figure size 640x480 with 2 axes>
```

(Maybe) Fixes Issue #9372 

Anything else that should go into the repr? Maybe dpi?

Could also do `__str__ == __repr__` seeing as they're nearly identical?

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [x] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -273,6 +273,14 @@ class Figure(Artist):
     def __str__(self):
         return "Figure(%gx%g)" % tuple(self.bbox.size)
 
+    def __repr__(self):
+        return "<{clsname} size {h:g}x{w:g} with {naxes} ax{ax_plural}>".format(
+            clsname=self.__class__.__name__,
+            h=self.bbox.size[0], w=self.bbox.size[1],
+            naxes=len(self.axes),
+            ax_plural='is' if len(self.axes) == 1 else 'es'
```
Comment (anntzer) [inline lib/matplotlib/figure.py]: The singular of Axes is Axes (... for matplotlib, Axis is something completely different (sad but true)).
TBH I would just have the suptitle and be done with it.

Comment (dstansby) [reply lib/matplotlib/figure.py]: Yeah, I think just saying "with one axes" is probably fine and technically more accurate, would it be possible to change that @richardjgowers ?

Comment (richardjgowers) [reply lib/matplotlib/figure.py]: @dstansby ok I've changed this

File: lib/matplotlib/figure.py [hunk 2]
```diff
@@ -273,6 +273,13 @@ class Figure(Artist):
     def __str__(self):
         return "Figure(%gx%g)" % tuple(self.bbox.size)
 
+    def __repr__(self):
+        return "<{clsname} size {h:g}x{w:g} with {naxes} axes>".format(
```
Comment (QuLogic) [inline lib/matplotlib/figure.py]: Capital Axes, though, referring to the class.

Comment (richardjgowers) [reply lib/matplotlib/figure.py]: done

**Pull request merged**